src/commons/sideContent/SideContentContestVoting.tsx

```diff
@@ -49,30 +30,172 @@ const SideContentContestVoting: React.FunctionComponent<SideContentContestVoting
+            'tier-a': 7,
+            'tier-s': 10
+          };
+          handleVotingSubmissionChange(contestEntry.submission_id, tierMapping[tierElement.id]);
+          console.log(tierMapping[tierElement.id]);
+        } else {
+          handleVotingSubmissionChange(contestEntry.submission_id, 0);
+        }
+      },
+    [handleVotingSubmissionChange]
+  );
+
+  const handleDragOver: React.DragEventHandler = e => {
+    console.log('Drag Over');
+    e.preventDefault();
+    e.dataTransfer.dropEffect = 'move';
+  };
+
+  const createPlaceholder = (): HTMLElement => {
+    console.log('create placeholder');
+    const el = document.createElement('div');
+    el.classList.add('placeholder');
+    setPlaceholder(el); // update the placeholder state with the new element
+    return el;
+  };
+
+  const handleDragEnter = useCallback(
+    (e: React.DragEvent): void => {
+      console.log('Drag Enter');
+      if (e.currentTarget.classList.contains('item-container')) {
+        if (!placeholder) {
+          setPlaceholder(createPlaceholder());
+        }
+
+        let insertBeforeElement: HTMLElement | null = null;
+
+        for (const child of Array.from((e.currentTarget as HTMLElement).children)) {
+          if (child === currentDraggedItem || child === placeholder) continue;
+
+          const childRect = (child as HTMLElement).getBoundingClientRect();
+          const childCenter = (childRect.left + childRect.right) / 2;
+
+          if (e.clientX < childCenter) {
+            insertBeforeElement = child as HTMLElement;
+            break;
+          }
+        }
+        if (insertBeforeElement && placeholder) {
+          e.currentTarget.insertBefore(placeholder as Node, insertBeforeElement);
+        } else if (placeholder) {
+          e.currentTarget.appendChild(placeholder as Node);
+        }
+      }
+    },
+    [placeholder, currentDraggedItem]
+  );
+
+  const handleDragLeave = useCallback(
+    (e: React.DragEvent): void => {
+      console.log('Drag Leave');
+      const container = e.currentTarget as HTMLElement;
+      if (
+        container.classList.contains('item-container') &&
+        !currentDraggedItem?.contains(e.relatedTarget as Node) &&
+        !placeholder?.contains(e.relatedTarget as Node) &&
+        container.contains(placeholder as Node)
+      ) {
+        container.removeChild(placeholder as Node);
+      }
+    },
+    [placeholder, currentDraggedItem]
+  );
+
+  const handleDrop = useCallback(
+    (e: React.DragEvent): void => {
+      console.log('Drop');
+      e.preventDefault();
+      const container = (e.currentTarget as HTMLElement).closest('.item-container');
+      if (container && container.contains(placeholder)) {
+        container.insertBefore(currentDraggedItem as Node, placeholder as Node);
+        container.removeChild(placeholder as Node);
+      }
+    },
+    [currentDraggedItem, placeholder]
+  );
+
+  const tierBoard = ['S', 'A', 'B', 'C', 'D'].map(tier => (
```

Comment:
I think it would be good if we move this array to the top-level as a read-only constant.

```tsx
const TIERS = ['S', 'A', 'B', 'C', 'D'] as const; // Outside the component

const SideContentContestVoting // ...
  // ...
  const tierBoard = TIERS.map(tier => (
  // ...
```

Reply:
Resolved in 991843b751fb007cdd74aa7fd0e425bc8e1103da, closing this thread.